lib/query.js

```diff
@@ -44,6 +48,47 @@ export const isNumber = node => {
     }
 }
 
+export const isInteger = node => {
+    return isNumber(node) && Number.isInteger(Number(node.value))
+}
+
+export const isConstantFraction = node => isFraction(node) && node.args.every(arg => isNumber(arg))
+export const isIntegerFraction = node => isFraction(node) && node.args.every(arg => isInteger(arg))
+
+export const isPolynomial = (node) => {
+    return isAdd(node) && node.args.every(isPolynomialTerm)
+}
+
+export const isVariableFactor = (node) =>
+    isIdentifier(node) ||
+    isPow(node) && isIdentifier(node.args[0]) &&
+    (isNumber(node.args[1]) || isVariableFactor(node.args[1]))
```

Comment:
Can you indent this line so that it's clearer that this depends on the the node being a power?